src/Core/functions.php

```diff
@@ -452,6 +452,11 @@ function toInt(mixed $value): ?int
             return $value;
         }
         if (is_string($value)) {
+            $value = trim($value);
+            if (preg_match('/0+[^0]+/', $value)) {
```

Comment:
```suggestion
            if (preg_match(/^0+[^0]{1}/, $value)) {
```

This would be more efficient at it would stop after matching the 1st non-zero character, which suffices for our needs.